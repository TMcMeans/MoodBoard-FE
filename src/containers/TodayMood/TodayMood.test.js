import React from 'react';
import { shallow } from 'enzyme';
import { TodayMood, mapStateToProps } from './TodayMood';

describe('TodayMood', () => {
  let journalMood = { entry_text: 'I am sad', tones: 'anger' };
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<TodayMood journal={journalMood} />);
  });

  it('should render a snapshot with anger tone passed in', () => {
    let mockJournal = { entry_text: 'I am sad', tones: 'anger' };
    wrapper = shallow(<TodayMood journal={mockJournal} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a snapshot with sadness tone passed in', () => {
    let mockJournal = { entry_text: 'I am sad', tones: 'sadness' };
    wrapper = shallow(<TodayMood journal={mockJournal} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a snapshot with tentative tone passed in', () => {
    let mockJournal = { entry_text: 'I am sad', tones: 'tentative' };
    wrapper = shallow(<TodayMood journal={mockJournal} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a snapshot with fear tone passed in', () => {
    let mockJournal = { entry_text: 'I am sad', tones: 'fear' };
    wrapper = shallow(<TodayMood journal={mockJournal} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a snapshot with joy tone passed in', () => {
    let mockJournal = { entry_text: 'I am sad', tones: 'joy' };
    wrapper = shallow(<TodayMood journal={mockJournal} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a snapshot with analytical tone passed in', () => {
    let mockJournal = { entry_text: 'I am sad', tones: 'analytical' };
    wrapper = shallow(<TodayMood journal={mockJournal} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a snapshot with confident tone passed in', () => {
    let mockJournal = { entry_text: 'I am sad', tones: 'confident' };
    wrapper = shallow(<TodayMood journal={mockJournal} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('mapStateToProps should return a journal object', () => {
    const mockState = {
      journal: {
        entry_text: 'I am so sad.',
        tones: 'sadness'
      }
    };

    const expected = { entry_text: 'I am so sad.', tones: 'sadness' };

    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps.journal).toEqual(expected);
  });
});
