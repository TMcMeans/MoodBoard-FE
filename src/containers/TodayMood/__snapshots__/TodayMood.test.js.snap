// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodayMood should render a snapshot with analytical tone passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodayMood
    journal={
      Object {
        "entry_text": "I am sad",
        "tones": "analytical",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Logo />,
        <h2
          className="today-title"
        >
          Today's Mood
        </h2>,
        <h1 />,
        <p
          className="message"
        >
          Based on your mood today,
          <br />
        </p>,
        undefined,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Today's Mood",
          "className": "today-title",
        },
        "ref": null,
        "rendered": "Today's Mood",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Based on your mood today,",
            <br />,
            undefined,
          ],
          "className": "message",
        },
        "ref": null,
        "rendered": Array [
          "Based on your mood today,",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          undefined,
        ],
        "type": "p",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Logo />,
          <h2
            className="today-title"
          >
            Today's Mood
          </h2>,
          <h1 />,
          <p
            className="message"
          >
            Based on your mood today,
            <br />
          </p>,
          undefined,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Today's Mood",
            "className": "today-title",
          },
          "ref": null,
          "rendered": "Today's Mood",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Based on your mood today,",
              <br />,
              undefined,
            ],
            "className": "message",
          },
          "ref": null,
          "rendered": Array [
            "Based on your mood today,",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            undefined,
          ],
          "type": "p",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`TodayMood should render a snapshot with anger tone passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodayMood
    journal={
      Object {
        "entry_text": "I am sad",
        "tones": "anger",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Logo />,
        <h2
          className="today-title"
        >
          Today's Mood
        </h2>,
        <h1 />,
        <p
          className="message"
        >
          Based on your mood today,
          <br />
        </p>,
        undefined,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Today's Mood",
          "className": "today-title",
        },
        "ref": null,
        "rendered": "Today's Mood",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Based on your mood today,",
            <br />,
            undefined,
          ],
          "className": "message",
        },
        "ref": null,
        "rendered": Array [
          "Based on your mood today,",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          undefined,
        ],
        "type": "p",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Logo />,
          <h2
            className="today-title"
          >
            Today's Mood
          </h2>,
          <h1 />,
          <p
            className="message"
          >
            Based on your mood today,
            <br />
          </p>,
          undefined,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Today's Mood",
            "className": "today-title",
          },
          "ref": null,
          "rendered": "Today's Mood",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Based on your mood today,",
              <br />,
              undefined,
            ],
            "className": "message",
          },
          "ref": null,
          "rendered": Array [
            "Based on your mood today,",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            undefined,
          ],
          "type": "p",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`TodayMood should render a snapshot with confident tone passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodayMood
    journal={
      Object {
        "entry_text": "I am sad",
        "tones": "confident",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Logo />,
        <h2
          className="today-title"
        >
          Today's Mood
        </h2>,
        <h1 />,
        <p
          className="message"
        >
          Based on your mood today,
          <br />
        </p>,
        undefined,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Today's Mood",
          "className": "today-title",
        },
        "ref": null,
        "rendered": "Today's Mood",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Based on your mood today,",
            <br />,
            undefined,
          ],
          "className": "message",
        },
        "ref": null,
        "rendered": Array [
          "Based on your mood today,",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          undefined,
        ],
        "type": "p",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Logo />,
          <h2
            className="today-title"
          >
            Today's Mood
          </h2>,
          <h1 />,
          <p
            className="message"
          >
            Based on your mood today,
            <br />
          </p>,
          undefined,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Today's Mood",
            "className": "today-title",
          },
          "ref": null,
          "rendered": "Today's Mood",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Based on your mood today,",
              <br />,
              undefined,
            ],
            "className": "message",
          },
          "ref": null,
          "rendered": Array [
            "Based on your mood today,",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            undefined,
          ],
          "type": "p",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`TodayMood should render a snapshot with fear tone passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodayMood
    journal={
      Object {
        "entry_text": "I am sad",
        "tones": "fear",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Logo />,
        <h2
          className="today-title"
        >
          Today's Mood
        </h2>,
        <h1 />,
        <p
          className="message"
        >
          Based on your mood today,
          <br />
        </p>,
        undefined,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Today's Mood",
          "className": "today-title",
        },
        "ref": null,
        "rendered": "Today's Mood",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Based on your mood today,",
            <br />,
            undefined,
          ],
          "className": "message",
        },
        "ref": null,
        "rendered": Array [
          "Based on your mood today,",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          undefined,
        ],
        "type": "p",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Logo />,
          <h2
            className="today-title"
          >
            Today's Mood
          </h2>,
          <h1 />,
          <p
            className="message"
          >
            Based on your mood today,
            <br />
          </p>,
          undefined,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Today's Mood",
            "className": "today-title",
          },
          "ref": null,
          "rendered": "Today's Mood",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Based on your mood today,",
              <br />,
              undefined,
            ],
            "className": "message",
          },
          "ref": null,
          "rendered": Array [
            "Based on your mood today,",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            undefined,
          ],
          "type": "p",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`TodayMood should render a snapshot with joy tone passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodayMood
    journal={
      Object {
        "entry_text": "I am sad",
        "tones": "joy",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Logo />,
        <h2
          className="today-title"
        >
          Today's Mood
        </h2>,
        <h1 />,
        <p
          className="message"
        >
          Based on your mood today,
          <br />
        </p>,
        undefined,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Today's Mood",
          "className": "today-title",
        },
        "ref": null,
        "rendered": "Today's Mood",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Based on your mood today,",
            <br />,
            undefined,
          ],
          "className": "message",
        },
        "ref": null,
        "rendered": Array [
          "Based on your mood today,",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          undefined,
        ],
        "type": "p",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Logo />,
          <h2
            className="today-title"
          >
            Today's Mood
          </h2>,
          <h1 />,
          <p
            className="message"
          >
            Based on your mood today,
            <br />
          </p>,
          undefined,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Today's Mood",
            "className": "today-title",
          },
          "ref": null,
          "rendered": "Today's Mood",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Based on your mood today,",
              <br />,
              undefined,
            ],
            "className": "message",
          },
          "ref": null,
          "rendered": Array [
            "Based on your mood today,",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            undefined,
          ],
          "type": "p",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`TodayMood should render a snapshot with sadness tone passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodayMood
    journal={
      Object {
        "entry_text": "I am sad",
        "tones": "sadness",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Logo />,
        <h2
          className="today-title"
        >
          Today's Mood
        </h2>,
        <h1 />,
        <p
          className="message"
        >
          Based on your mood today,
          <br />
        </p>,
        undefined,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Today's Mood",
          "className": "today-title",
        },
        "ref": null,
        "rendered": "Today's Mood",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Based on your mood today,",
            <br />,
            undefined,
          ],
          "className": "message",
        },
        "ref": null,
        "rendered": Array [
          "Based on your mood today,",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          undefined,
        ],
        "type": "p",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Logo />,
          <h2
            className="today-title"
          >
            Today's Mood
          </h2>,
          <h1 />,
          <p
            className="message"
          >
            Based on your mood today,
            <br />
          </p>,
          undefined,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Today's Mood",
            "className": "today-title",
          },
          "ref": null,
          "rendered": "Today's Mood",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Based on your mood today,",
              <br />,
              undefined,
            ],
            "className": "message",
          },
          "ref": null,
          "rendered": Array [
            "Based on your mood today,",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            undefined,
          ],
          "type": "p",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`TodayMood should render a snapshot with tentative tone passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodayMood
    journal={
      Object {
        "entry_text": "I am sad",
        "tones": "tentative",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Logo />,
        <h2
          className="today-title"
        >
          Today's Mood
        </h2>,
        <h1 />,
        <p
          className="message"
        >
          Based on your mood today,
          <br />
        </p>,
        undefined,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Today's Mood",
          "className": "today-title",
        },
        "ref": null,
        "rendered": "Today's Mood",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Based on your mood today,",
            <br />,
            undefined,
          ],
          "className": "message",
        },
        "ref": null,
        "rendered": Array [
          "Based on your mood today,",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          undefined,
        ],
        "type": "p",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Logo />,
          <h2
            className="today-title"
          >
            Today's Mood
          </h2>,
          <h1 />,
          <p
            className="message"
          >
            Based on your mood today,
            <br />
          </p>,
          undefined,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Today's Mood",
            "className": "today-title",
          },
          "ref": null,
          "rendered": "Today's Mood",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Based on your mood today,",
              <br />,
              undefined,
            ],
            "className": "message",
          },
          "ref": null,
          "rendered": Array [
            "Based on your mood today,",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            undefined,
          ],
          "type": "p",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
